### PR TITLE
make bind work with keyword functions

### PR DESCRIPTION
**Pull request by jw3126**
I like the elegant implementation in your package a lot. Here is some better keyword support.

### CODE REVIEW

File: src/Curry.jl [hunk 1]
```diff
@@ -62,19 +62,20 @@ b(", world")
...
 end
 
-function (c::Bind)(args...)
-    c.f(interleave(c.a, args)...)
+function (c::Bind)(args...; kw...)
+    c.f(interleave(c.a, args)...; c.k..., kw...)
 end
 
 """
 `bind(f, a, b)` is equivalent to Bind(f, (a, b))
```
Comment (goretkin) [inline src/Curry.jl]: Could you amend this docstring to include a keyword arg example?

File: src/Curry.jl [hunk 2]
```diff
@@ -83,7 +84,8 @@ bind(f, args...) = Bind(f, args)
 macro bind(ex)
     ex.head == :call || error()
     # `ex.args[1]` is the function and `ex.args[2:end]` are the positional arguments
-    return :(bind($(map(esc, ex.args)...)))
+    TODO_HANDLE_KW = ()
```
Comment (goretkin) [inline src/Curry.jl]: Does this serve the same purpose as a `TODO`comment?

**Comment by goretkin** [comment]
Thanks for the interest and for the improvements!

File: src/Curry.jl (hunk 2 above)
Comment (jw3126) [reply src/Curry.jl]: yeah sorry it is a leftover from some other thing I tried.

**Pull request merged**